Finish My project with Pom and create tests

the7thsaint opened the pull request.

the7thsaint committed:
create pageObject
create .gitignoreFile
changes in pom.xml

the7thsaint committed:
Create Tests

egorvas commented on src/test/java/org/example/BaseTest.java:
```diff
@@ -0,0 +1,32 @@
+
+    @Before
+    public void startUp() {
+        System.setProperty("webdriver.chrome.driver","/Users/pavelvinogradov/Documents/bin/chromedriver");
```
**Нужно исправить:** у меня на компьютере этого файла нет, поэтому твои тесты не запускаются. Пожалуйста, используй способ, который будет работать на всех компьютерах.

the7thsaint replied:
проправил

egorvas commented on src/test/java/org/example/TestCreateOrder.java:
```diff
@@ -0,0 +1,42 @@
+
+
+    @Test
+    public void TestOrderWithHeaderButton(){
```
**Нужно исправить:** функции java всегда с маленькой буквы.

the7thsaint replied:
поправил)

egorvas commented on src/test/java/org/example/TestCreateOrder.java:
```diff
@@ -0,0 +1,42 @@
+        UserInfoOrderPageElements objUserInfo = new UserInfoOrderPageElements(driver);
+        RentInfoOrderPageElements objInfoOrdred = new RentInfoOrderPageElements(driver);
+
+        objelements.pushAcceptCookieButton();
```
**Нужно исправить:** то что не связано с логикой тесты нужно выносить в after/before функции

the7thsaint replied:
поправил

egorvas commented on src/test/java/org/example/TestCreateOrderParametrized.java:
```diff
@@ -0,0 +1,51 @@
+        };
+    }
+    @Test
+    public void TestOrder(){
```
**Нужно исправить:** это очень плохое название теста. Название должно описывать что именно происходит в тесте.

the7thsaint replied:
поправил

egorvas commented on src/test/java/org/example/TestCreateOrderParametrized.java:
```diff
@@ -0,0 +1,51 @@
+
+
+@RunWith(Parameterized.class)
+public class TestCreateOrderParametrized extends BaseTest{
```
**Нужно исправить:** совершенно нет смысла дублировать тесты

the7thsaint replied:
Тут тест задублен с точки зрения того, что если вдруг понадобилось бы прогнать 1 раз с какой то инфой - пользователь мог его прогнать 1 раз

the7thsaint replied:
поправил

egorvas commented on src/test/java/org/example/TestListOfQuestionsAboutImportant.java:
```diff
@@ -0,0 +1,15 @@
+    @Test
+    public void ListsEquals(){
+        MainPageElements objElements = new MainPageElements(driver);
+        objElements.equalsListsOfQuestionsAboutImportant();
```
**Нужно исправить:** тебе нужно написать такие тесты, чтобы каждый элемент FAQ проверялся по отдельности. Иначе если ошибка будет в 1ом элементе FAQ, то тесты по остальным не пройдут. Рекомендую использовать параметризацию, для упрощения кода.

the7thsaint committed:
Refactor tests naming
Create parametrised test for List Question Abount important
add to Order test parametrised test with footer button
edit Web driver configuration
add to @Before click to accept cookie button
